audio-worklet/src/bitcrusher-demo.js

```diff
@@ -0,0 +1,140 @@
+/*
+  Copyright 2017 Google Inc.
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+class BitcrusherDemo {
+  /**
+   * Initalizes and modifies bitcrusher settings in response to GUI input.
+   * @param  {Number} bitDepth the number of bits for each sample
+   * @param  {Number} reduction the amount of sample rate reduction to apply
+   * @param  {Number} gain the volume of the output
+   */
+  constructor(bitDepth, reduction, gain) {
+    this.context_ = new AudioContext();
+    this.gainNode_ = new GainNode(this.context_);
+    this.gainNode_.gain.value = gain || 0.5;
+
+    this.bitcrusher_ = new Bitcrusher(this.context_, {
+      inputChannels: 1,
+      outputChannels: 1,
+      bitDepth: bitDepth || 24,
+      reduction: reduction || 1
+    });
+
+    this.bitcrusher_.output.connect(this.gainNode_);
+    this.gainNode_.connect(this.context_.destination);
+
+    const songUrl = 'audio/it-from-bit.mp3';
+
+    // Fetch song from server and decode into buffer.
+    fetch(songUrl).then(response => response.arrayBuffer().then(song => {
+      this.context_.decodeAudioData(song).then(buffer => {
+        this.songBuffer = buffer;
+        this.sourceButton_.enable();
+      });
```

Comment:
We had talked earlier about changing this to something like
```
fetch().then(...)
       .then(...)
       .then(...)
```
or, as Hongchan suggested, using `await`.